src/btree/row_modify.c

```diff
@@ -133,8 +133,8 @@ __wt_row_modify(WT_CURSOR_BTREE *cbt, const WT_ITEM *key, const WT_ITEM *value,
             WT_ASSERT(session,
               !WT_IS_HS(S2BT(session)->dhandle) ||
                 (*upd_entry == NULL ||
-                  ((*upd_entry)->type == WT_UPDATE_TOMBSTONE && (*upd_entry)->txnid == WT_TS_NONE &&
-                    (*upd_entry)->start_ts == WT_TS_NONE)) ||
+                  ((*upd_entry)->type == WT_UPDATE_TOMBSTONE &&
+                    (*upd_entry)->txnid == WT_TXN_NONE && (*upd_entry)->start_ts == WT_TS_NONE)) ||
```

Comment:
Nice catch - do we need to be concerned about this? It looks like a bug regardless of the backport here?

Reply:
Yeah, it looks like a bug which I decided to just include here. I think it's actually ok because `WT_TS_NONE` and `WT_TXN_NONE` are actually the same value in the codebase, it's just confusing to read in the condition.